RSS Reader/RSS Reader/View Controllers/Web Browser/RSSRWebBrowserController.m

```diff
@@ -0,0 +1,212 @@
+    }
+    return self;
+}
+
+- (void)dealloc {
+    [_webView release];
+    [_request release];
+    [_backButton release];
+    [_forwardButton release];
+    [_reloadButton release];
+    [_stopButton release];
+    [super dealloc];
+}
+
+
+#pragma mark - View controller life cycle
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    self.navigationController.navigationBar.barTintColor = UIColor.whiteColor;
+    
+    [self configureNavigationController];
+    [self configureToolbarItems];
+    [self disableControlButtons];
+    
+    self.webView.scrollView.delegate = self;
+    self.webView.navigationDelegate = self;
+
+    [self startLoading];
+}
+
+
+
+- (void)configureToolbarItems {
+    UIBarButtonItem *safariButton = [[UIBarButtonItem alloc] initWithImage:UIImage.RSSRExplore
+                                                               style:UIBarButtonItemStylePlain
+                                                              target:self
+                                                              action:@selector(openInSafari)];
+    UIBarButtonItem *flexibleSpace = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemFlexibleSpace
+                                                                                   target:nil
+                                                                                   action:nil];
+    
+    [self setToolbarItems:@[self.backButton, flexibleSpace, self.forwardButton, flexibleSpace, self.reloadButton, flexibleSpace, self.stopButton, flexibleSpace, safariButton]];
+    
+    [safariButton release];
+    [flexibleSpace release];
+}
+
+- (void)configureNavigationController {
+    [self.navigationController setHidesBarsOnSwipe:NO];
+    [self.navigationController setToolbarHidden:NO];
+    self.navigationItem.title = [NSURL URLWithString:self.request.URL.absoluteString].host;
+    
+    UIBarButtonItem *doneBackButton = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemDone
+                                                                                    target:self
+                                                                                    action:@selector(popViewController)];
+    self.navigationItem.leftBarButtonItem = doneBackButton;
+    [doneBackButton release];
+}
+
+- (void)popViewController {
+    [self.navigationController popViewControllerAnimated:YES];
+}
+
+- (void)startLoading {
+    [self.webView loadRequest:self.request];
+}
+
+- (void)disableControlButtons {
+    self.backButton.enabled = NO;
+    self.forwardButton.enabled = NO;
+    self.stopButton.enabled = NO;
+    self.reloadButton.enabled = NO;
+}
+
+
+#pragma mark - Hide/show toolbar and navigationbar
+
+- (void)scrollViewWillBeginDragging:(UIScrollView *)scrollView {
+    self.lastContentOffset = scrollView.contentOffset.y;
+}
+
+- (void)scrollViewDidScroll:(UIScrollView *)scrollView {
+    if (scrollView.contentOffset.y <= self.lastContentOffset) {
```

Comment:
There is too much code here which almost have no difference. We can improve it the next way.

- (void)scrollViewDidScroll:(UIScrollView *)scrollView {
     bool didScroll = scrollView.contentOffset.y <= self.lastContentOffset;
    [self.navigationController setToolbarHidden:!didScroll animated:YES];
    [self.navigationController setNavigationBarHidden:!didScroll animated:YES];
 }

This allows you to escape from nesting, makes your code shorter and easy to read.

Reply:
Done